frontend/src/middleware.ts

```diff
@@ -0,0 +1,13 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+
+export function middleware(request: NextRequest) {
```

Comment:
Please justify the need to both have this middleware and the isAuth context.

Reply:
Sure. Middleware eliminates the need to check for authentication status on each protected resource component. Instead of repeatedly checking if isAuthenticated with the useAuth in each component/layout we could just add a matcher that would redirect them to login page if the login cookie isn't present. Also middleware could also match api endpoints whereas a client-side useAuth cannot